src/cargo/core/features.rs

```diff
@@ -286,6 +292,7 @@ impl Edition {
             Edition2015 => false,
             Edition2018 => true,
             Edition2021 => true,
+            Edition2024 => false,
```

Comment:
While there currently is [`rust_2024_compatibility`](https://github.com/rust-lang/rust/blob/187b8131d4f760f856b214fce34534903276f2ef/compiler/rustc_span/src/edition.rs#L57) lint group, I set it to [`false`](https://github.com/Muscraft/cargo/blob/b3481b7bf36c0405238cd2404ae8bdd5eaeba65e/src/cargo/core/features.rs#L295) as there are no lints in the group currently (from what I can tell). This was how `Edition2021` originally was before it was [updated to be `true`](https://github.com/rust-lang/cargo/pull/9588/files#diff-7b081a836e71ae12b28f42ba8f09384036be9ae8ee479b4849f43a8522e94b37R219).